idseq_pipeline/commands/accessionid2seq_functions.py

```diff
@@ -234,29 +237,33 @@ def get_sequence_for_thread(error_flags, accession_info, accession_id, nt_loc_di
 
 def get_sequence_by_accession_id_ntf(accession_id, nt_loc_dict, ntf):
     ref_seq = ''
+    seq_name = ''
     entry = nt_loc_dict.get(accession_id)
     if entry:
-        range_start = entry[0]+entry[1]
-        seq_len = entry[2]
+        range_start = entry[0]
+        seq_len = entry[1] + entry[2]
         ntf.seek(range_start, 0)
-        ref_seq = ntf.read(seq_len).replace("\n", "")
-    return ref_seq
+        (seq_name, ref_seq) = ntf.read(seq_len).split("\n", 1)
+        ref_seq = ref_seq.replace("\n", "")
+        seq_name = seq_name.split(" ", 1)[1]
+    return (ref_seq, seq_name)
 
 def get_sequence_by_accession_id_s3(accession_id, nt_loc_dict, nt_bucket, nt_key):
     seq_len = 0
+    seq_name = ''
     entry = nt_loc_dict.get(accession_id)
     if entry:
-        range_start = entry[0]+entry[1]
-        seq_len = entry[2]
+        (range_start, name_length, seq_len) = entry
```

Comment:
i didn't realize you knew the name length already!   then the simplest solution would have been to simply add the name length to all the seek offsets when you read the accession file, and keep the name at the beginning of the accession file

Reply:
but there is something to be said about the file looking cleaner this way